### PR TITLE
*: improve time codec.

### PR DESCRIPTION
**Pull request by coocood**
Use a more simple and efficient way to encode Time types.

### CODE REVIEW

File: mysql/time.go [hunk 1]
```diff
@@ -346,6 +285,59 @@ func (t Time) RoundFrac(fsp int) (Time, error) {
 	return Time{Time: nt, Type: t.Type, Fsp: fsp}, nil
 }
 
+// ToBin encodes Time to 12 bytes binary data.
+func (t Time) ToBin() []byte {
```
Comment (shenli) [inline mysql/time.go]: Any reference or comment for the encode/decode algorithm?

File: mysql/time.go [hunk 2]
```diff
@@ -30,6 +31,7 @@ var (
 	ErrInvalidTimeFormat = errors.New("invalid time format")
 	ErrInvalidYearFormat = errors.New("invalid year format")
 	ErrInvalidYear       = errors.New("invalid year")
+	ErrInvalidBinary     = errors.New("invalid binary")
```
Comment (shenli) [inline mysql/time.go]: "invalid time binary" will be more helpful for debug.

**Comment by coocood** [comment]
@shenli @BusyJay PTAL

**Comment by shenli** [comment]
LGTM
PTAL @zimulala @tiancaiamao @hanfei19910905

File: mysql/time.go [hunk 3]
```diff
@@ -285,47 +284,55 @@ func (t Time) RoundFrac(fsp int) (Time, error) {
...
 
-// FromBin decodes 12 bytes binary data to Time.
-func (t *Time) FromBin(bin []byte) error {
-	if len(bin) != 12 {
-		return ErrInvalidBinary
-	}
-	intPart := int(binary.BigEndian.Uint64(bin))
-	fracPart := binary.BigEndian.Uint32(bin[8:])
-	if intPart == 0 && fracPart == 0 {
+// FromPackedUint decodes Time from an packed uint64 value.
```
Comment (zimulala) [inline mysql/time.go]: an -> a ?

**Comment by zimulala** [comment]
LGTM

**Pull request merged**